### ISSUE TITLE
[BUG] split_trajectories does not produce the right output

### ISSUE THREAD

**Issue by AlexandreBrown** [pytorch/rl#2659]
## Describe the bug
When using [`split_trajectories`](https://pytorch.org/rl/stable/reference/generated/torchrl.collectors.utils.split_trajectories.html) to split a tensordict of shape (150,) into a tensordict of shape (3,50), I get a tensordict of shape (6, 50).    

In my setup, 1 episode = 50 steps, I did a test with 3 episodes this gives us a tensordict of shape (3 x 50 = 150).  
When looking at the `tensordict["collector"]["traj_ids"]` we can see the different trajectories and they make sense. 

## To Reproduce
1. create a sync data collector with -1 total frames
2. 
```python
max_nb_frames = self.eval_num_rollouts * self.env_max_frames_per_traj

assert (
    max_nb_frames % self.eval_frames_per_batch == 0
), "Number of frames per batch should be a divisor of the total number of frames"

nb_iters = max_nb_frames // self.eval_frames_per_batch

rollouts_data = []
for i, eval_data_batch in enumerate(
    tqdm(self.eval_data_collector, "Evaluation", total=nb_iters)
):
    if eval_data_batch.ndim == 2:
        eval_data_batch = eval_data_batch.flatten()
    eval_data_batch = eval_data_batch.select(
        ("collector", "traj_ids"),
        ("next", "reward"),
        ("next", "done"),
        "action",
        "pixels",
        "pixels_masked",
        strict=False,
    ).cpu()
    rollouts_data.append(eval_data_batch)

    if i + 1 == nb_iters:
        break

rollouts_data_concatenated = torch.cat(rollouts_data, dim=0)

output = split_trajectories(rollouts_data_concatenated)
```  
Or you can load the torch tensors that I used to observe the bug : 
1. `rollout data`: https://filebin.net/zsyi73np2q49anr3/rollouts_data.pt
2. `rollouts_data_concatenated`: https://filebin.net/zsyi73np2q49anr3/rollouts_data_concatenated.pt
3. `output` : https://filebin.net/zsyi73np2q49anr3/output.pt  

## Expected behavior

A tensordict of shape (B, T) where T = the biggest trajectory, in my case B = 3 and T = 50.

## Actual behavior  
A tensordict of shape (6, 50)

## System info

Describe the characteristic of your environment:
 * Describe how the library was installed (pip, source, ...): pip
 * Python version: 3.10
```python
print(torchrl.__version__, numpy.__version__, sys.version, sys.platform)
0.6.0 1.26.4 3.10.16 | packaged by conda-forge | (main, Dec  5 2024, 14:16:10) [GCC 13.3.0] linux
```

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have read the [documentation](https://github.com/pytorch/rl/tree/main/docs/) (**required**)
- [x] I have provided a minimal working example to reproduce the bug (**required**)

**Comment by vmoens** [comment]
> In my setup, 1 episode = 50 steps, I did a test with 3 episodes this gives us a tensordict of shape (3 x 50 = 150).
When looking at the tensordict["collector"]["traj_ids"] we can see the different trajectories and they make sense.

Hmm not sure I can see that
```
rollouts_data[0]["collector", "traj_ids"]
Out[7]: 
tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1])
rollouts_data[1]["collector", "traj_ids"]
Out[8]: 
tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
        1, 1])
rollouts_data[2]["collector", "traj_ids"]
Out[9]: 
tensor([2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
        2, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
        3, 3])
```
The trajs go 0 -1 - 0 - 1 - 2 - 3
It should be strictly increasing. Any idea what could be going on?

If I correct them split_trajectories works fine
```
rollouts_data_concatenated["collector", "traj_ids"][50:] += 2
output = split_trajectories(rollouts_data_concatenated)
output["collector", "traj_ids"]
Out[15]: 
tensor([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0],
        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1],
        [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
         2],
        [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
         3],
        [4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4, 4,
         4],
        [5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5, 5,
         5]])
```

**Comment by AlexandreBrown** [comment]
@vmoens I see, I'm using Maniskill3 (2 parallel env) which is why it is outputting traj_ids [0,...,0,...,1,...,1] then [0,...,0,...,1,...,1] in the next iteration.  
If sorting them to be strictly increasing fixes the issue then I believe this is more of a documentation issue so that people can sort their data correctly for `split_trajectories` (or should `split_trajectories` do it?)

**Comment by vmoens** [comment]
Can you elaborate on what format the data comes in?
I don't think it's split_traj responsibility of sorting the data, it'd be very hard to get something generic and efficient enough.

**Comment by AlexandreBrown** [comment]
The format is `(B, T)` where B = nb of batched envs and T = time.  
> I don't think it's split_traj responsibility of sorting the data, it'd be very hard to get something generic and efficient enough.  
  
Fair enough!

**Comment by vmoens** [comment]
This gives me the right result
```python
import torch
from torchrl.collectors.utils import split_trajectories

rollouts_data = torch.load("/Users/vmoens/Downloads/rollouts_data.pt", weights_only=False)
rollouts_data = [r.view(2, -1) for r in rollouts_data]
rollouts_data_concatenated = torch.cat(rollouts_data, -1)

output = split_trajectories(rollouts_data_concatenated)
print(output["collector", "traj_ids"])
```
so you just need to separate the batch and time dim (like you suggested above) and split_traj will know what to do
```
tensor([[0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0],
        [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2,
         2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0],
        [1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
         1, 1],
        [3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3, 3,
         3, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
         0, 0]])
```
